beacon-chain/core/epoch/epoch_processing.go

```diff
@@ -381,9 +381,19 @@ func ProcessHistoricalRootsUpdate(state state.BeaconState) (state.BeaconState, e
 	// Set historical root accumulator.
 	epochsPerHistoricalRoot := params.BeaconConfig().SlotsPerHistoricalRoot.DivSlot(params.BeaconConfig().SlotsPerEpoch)
 	if nextEpoch.Mod(uint64(epochsPerHistoricalRoot)) == 0 {
+		blockRoots := state.BlockRoots()
```

Comment:
This is going to be a big drag on performance, this whole piece of code does 4 copies